src/requests/misc.jl

```diff
@@ -5,47 +5,49 @@ end
+                    end
+                    if loc + a.args[1].fullspan <= offset <= loc + a.args[1].fullspan + a.args[2].span # Within name of the module, so return entire expression
+                        return Position(get_position_at(doc, loc)...), Position(get_position_at(doc, loc + a.span)...), Position(get_position_at(doc, loc + a.fullspan)...)
+                    end
+
+                    if loc + a.args[1].fullspan + a.args[2].fullspan <= offset <= loc + a.args[1].fullspan + a.args[2].fullspan + a.args[3].span # Within the body of a module
                         loc += a.args[1].fullspan + a.args[2].fullspan
                         for b in a.args[3].args
-                            if loc <= offset < loc + b.fullspan
-                                p1, p2, p3 = loc, loc + b.span, loc + b.fullspan
-                                break
+                            if loc <= offset <= loc + b.span
+                                return Position(get_position_at(doc, loc)...), Position(get_position_at(doc, loc + b.span)...), Position(get_position_at(doc, loc + b.fullspan)...)
                             end
                             loc += b.fullspan
                         end
-                    else
-                        p1, p2, p3 = loc, loc + a.span, loc + a.fullspan
+                    elseif loc + a.args[1].fullspan + a.args[2].fullspan + a.args[3].fullspan < offset < loc + a.args[1].fullspan + a.args[2].fullspan + a.args[3].fullspan + a.args[4].span # Within `end` of the module, so return entire expression
+                        return Position(get_position_at(doc, loc)...), Position(get_position_at(doc, loc + a.span)...), Position(get_position_at(doc, loc + a.fullspan)...)
                     end
                 elseif typof(a) === CSTParser.TopLevel
-                    p1, p2, p3 = loc, loc + a.span, loc + a.fullspan
                     for b in a.args
-                        if loc <= offset < loc + b.fullspan
-                            p1, p2, p3 = loc, loc + b.span, loc + b.fullspan
+                        if loc <= offset <= loc + b.span
+                            return Position(get_position_at(doc, loc)...), Position(get_position_at(doc, loc + b.span)...), Position(get_position_at(doc, loc + b.fullspan)...)
                         end
                         loc += b.fullspan
                     end
                 else
-                    p1, p2, p3 = loc, loc + a.span, loc + a.fullspan
+                    return Position(get_position_at(doc, loc)...), Position(get_position_at(doc, loc + a.span)...), Position(get_position_at(doc, loc + a.fullspan)...)
                 end
             end
             loc += a.fullspan
         end
     end
-    return Position(get_position_at(doc, p1)...), Position(get_position_at(doc, p2)...), Position(get_position_at(doc, p3)...)
+    return Position(0, 0), Position(0, 0), tdpp.position
```

Comment:
Could we try to get the next block here as well instead of returning the current cursor position?